packages/stark-ui/src/modules/input-mask-directives/directives/email-mask.directive.ts

```diff
@@ -13,100 +14,86 @@ const directiveName = "[starkEmailMask]";
 	selector: directiveName,
 	exportAs: "starkEmailMask",
-	providers: [STARK_EMAIL_MASK_VALUE_ACCESSOR]
+	providers: [STARK_EMAIL_MASK_NEW_VALUE_ACCESSOR]
 })
-export class StarkEmailMaskDirective extends MaskedInputDirective implements OnChanges {
-	/**
-	 * Whether to display the email mask in the input field.
-	 */
-	/* tslint:disable:no-input-rename */
-	@Input("starkEmailMask")
-	public maskConfig = true; // enabled by default
-
-	// Information about boolean coercion https://angular.io/guide/template-typecheck#input-setter-coercion
+export class StarkEmailMaskDirective extends TextMaskBaseDirective<IMask.MaskedPatternOptions, StarkTextMaskConfig> {
 	// tslint:disable-next-line:variable-name
 	public static ngAcceptInputType_maskConfig: BooleanInput;
 
-	/**
-	 * Class constructor
-	 * @param _renderer - Angular `Renderer2` wrapper for DOM manipulations.
-	 * @param _elementRef - Reference to the DOM element where this directive is applied to.
-	 * @param _compositionMode  - Injected token to control if form directives buffer IME input until the "compositionend" event occurs.
-	 */
+	// Information about boolean coercion https://angular.io/guide/template-typecheck#input-setter-coercion
+	/* tslint:disable:no-input-rename */
+	@Input("starkEmailMask")
+	public override maskConfig = true;
+
 	public constructor(
 		_renderer: Renderer2,
 		_elementRef: ElementRef,
+		_factory: IMaskFactory,
+		@Inject(PLATFORM_ID) _platformId: string,
 		@Optional() @Inject(COMPOSITION_BUFFER_MODE) _compositionMode: boolean
 	) {
-		super(_renderer, _elementRef, _compositionMode);
+		super(_renderer, _elementRef, _factory, _platformId, _compositionMode);
 	}
 
-	/**
-	 * Component lifecycle hook
-	 */
-	public override ngOnChanges(changes: SimpleChanges): void {
-		this.textMaskConfig = this.normalizeMaskConfig(this.maskConfig);
-
-		super.ngOnChanges(changes);
+	protected override defaultMask(): StarkTextMaskConfig {
+		return {
+			mask: DEFAULT_EMAIL_PATTERN,
+			blocks: {
+				name: { mask: /^[\w-\\.]+$/g },
+				domain: { mask: /^[\w-]+$/g },
+				tld: { mask: /^\w+$/g }
+			},
+			guide: true,
+			eager: false,
+			keepCharPositions: true,
+			placeholderChar: "_"
+		};
 	}
 
-	/**
-	 * Create a valid configuration to be passed to the MaskedInputDirective
-	 * @param maskConfig - The provided configuration via the directive's input
-	 */
-	public normalizeMaskConfig(maskConfig: boolean = true): Ng2TextMaskConfig {
-		// in case the directive is used without inputs: "<input type='text' starkEmailMask>" the maskConfig becomes an empty string ''
-		// therefore "undefined" or string values will also enable the mask
-		maskConfig = typeof maskConfig !== "boolean" ? true : maskConfig;
+	protected override normalizeMaskConfig(maskConfig: boolean | string | StarkTextMaskConfig, defaultMask: StarkTextMaskConfig): any {
+		const mask: StarkTextMaskConfig = this.mergeMaskConfig(maskConfig, defaultMask);
+		return {
+			...mask,
+			...defaultMask,
+			blocks: {
```

Comment:
This return should be reworked since it applies the same logic than `this.mergeMaskConfig(maskConfig, defaultMask)` just above.